manifests/tools/update_imagestream_annotations_from_pylock.py

```
@@ -0,0 +1,684 @@
+                shown = _git_show_first_existing(sha, rel_paths)
+            if shown is None:
+                print(
+                    f"skip {path.name} tag {idx}: no lockfile (tried worktree/git {'; '.join(rel_paths)})",
+                    file=sys.stderr,
+                )
+                continue
+            rel_used, text = shown
+            py_minor = _python_minor_from_dir(nb_dir)
+            try:
+                pkgs = load_packages_from_lockfile(text, rel_used, py_minor)
+            except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)